stx-genesis/build.rs

```diff
@@ -0,0 +1,102 @@
+use std::fmt::Write as FmtWrite;
+use std::fs;
+use std::path::Path;
+use std::{
+    env,
+    fs::File,
+    io::{BufRead, BufReader, Read, Write},
+};
+
+use libflate::deflate;
+use sha2::{Digest, Sha256};
+
+fn main() {
+    verify_genesis_integrity();
+    write_archives().expect("failed to write archives");
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=chainstate.txt.sha256");
+    println!("cargo:rerun-if-changed=chainstate.txt");
+}
+
+fn open_chainstate_file() -> File {
+    File::open("chainstate.txt").unwrap()
+}
+
+pub fn write_archives() -> std::io::Result<()> {
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    {
+        let chainstate_file = open_chainstate_file();
+        let reader = BufReader::new(chainstate_file);
+        let balances_file_path = Path::new(&out_dir).join("account_balances.gz");
+        let balances_file = File::create(balances_file_path)?;
+        let mut balances_encoder = deflate::Encoder::new(balances_file);
+
+        for line in reader
+            .lines()
+            .map(|line| line.unwrap())
+            .skip_while(|line| !line.eq(&"-----BEGIN STX BALANCES-----"))
+            // skip table header line "address,balance"
+            .skip(2)
```

Comment:
Just making sure -- `.skip(2)` will skip `-----BEGIN STX BALANCES-----` and `address,balance`, not `address,balance` and the first actual address and balance, right?

Reply:
Correct! I'd remove the table header, but I like the readability of a "csv" enclosed in PEM-like headers.